django_cloud_deploy/tests/unit/crash_handling/crash_handling_test.py

```diff
@@ -0,0 +1,82 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit test for django_cloud_deploy/crash_handling."""
+
+import unittest
+from unittest import mock
+from urllib import parse
+
+from django_cloud_deploy import crash_handling
+from django_cloud_deploy.cli import io
+
+
+@mock.patch('tempfile.mkstemp', return_value=(1, '/tmp/tmpfile'))
+@mock.patch('os.fdopen')
+class CrashHandlingTest(unittest.TestCase):
+    """Unit test for django_cloud_deploy.crash_handling."""
+
+    _COLLECTED_INFO = ('Django Cloud Deploy:', 'Gcloud Version',
+                       'Docker Version', 'Cloud SQL Proxy Version',
+                       'Python Version', 'Traceback', 'Platform',
+                       'Issue running command')
+
+    def assert_valid_issue_title(self, title, err):
+        self.assertIn(type(err).__name__, title)
+        self.assertIn(str(err), title)
+
+    def assert_valid_issue_body(self, body):
+        for keyword in self._COLLECTED_INFO:
+            self.assertIn(keyword, body)
+
+    def parse_body_and_title_from_url(self, url):
+        params = parse.parse_qs(parse.urlparse(url).query)
+        return params['title'][0], params['body'][0]
+
+    @mock.patch('webbrowser.open')
+    def test_create_issue(self, mock_open_browser, *unused_mocks):
+        test_io = io.TestIO()
+        test_io.answers.append('Y')  # Agree to open browser to create an issue
+        error = KeyError('Test KeyError')
+        try:
+            raise error
+        except KeyError as e:
+            crash_handling.handle_crash(e, 'command_fake', test_io)
+        self.assertEqual(mock_open_browser.call_count, 1)
+        url = mock_open_browser.call_args[0][0]
+        title, body = self.parse_body_and_title_from_url(url)
+        self.assert_valid_issue_title(title, error)
+        self.assert_valid_issue_body(body)
+
+    @mock.patch('webbrowser.open')
+    def test_not_creating_issue(self, mock_open_browser, *unused_mocks):
+        test_io = io.TestIO()
+        test_io.answers.append('N')  # Not creating Github issues
+        error = KeyError('Test KeyError')
+        try:
+            raise error
+        except KeyError as e:
+            crash_handling.handle_crash(e, 'command_fake', test_io)
+        mock_open_browser.assert_not_called()
+
+    @mock.patch('webbrowser.open')
+    def test_ignore_user_error(self, mock_open_browser, *unused_mocks):
+        test_io = io.TestIO()
+        test_io.answers.append('Y')  # Agree to open browser to create an issue
+        error = crash_handling.UserError('Test UserError')
+        with self.assertRaises(crash_handling.UserError):
+            try:
+                raise error
+            except KeyError as e:
+                crash_handling.handle_crash(e, 'command_fake', test_io)
+        mock_open_browser.assert_not_called()
```

Comment:
Can you assert that something was written to stderr?

Reply:
The logic of this test case was wrong. Now I assert KeyError is raised instead of asserting UserError is raised.

Reply:
But something should still be written to stderr, right?

Reply:
Don't we assume if an error is thrown it has to show on the stderr?

Reply:
Yeah, I misunderstood how this worked.